sql/src/main/java/io/crate/metadata/Schemas.java

```diff
@@ -158,25 +162,36 @@ public void clusterChanged(ClusterChangedEvent event) {
                 }
             }
 
+            for (String addedSchema : added) {
+                schemas.put(addedSchema, getCustomSchemaInfo(addedSchema));
+            }
+
             // update all existing schemas
             for (SchemaInfo schemaInfo : this) {
```

Comment:
@mfussenegger yes, for the case that a function introduces a new schema we want that schema immediately updated, so it can registers the new function.